### DISCUSSION TITLE
Issue with `router.refresh()` Not Updating Dropdown in Next.js 13

### DISCUSSION THREAD

**Discussion by vladimirzb** [vercel/next.js#54867] [Help] [answered]
### Summary

I'm working on a Next.js 13 project involving two client components: `AddFrequentIngreso` and `AddIngreso`. The former is responsible for adding frequent income sources to a database, while the latter should display these in a dropdown. I've tried using `router.refresh()` to update the dropdown in `AddIngreso` after adding a new income source via `AddFrequentIngreso`. However, this only updates other components on the page, not the dropdown in `AddIngreso`. I'm looking for guidance on why this is happening and how to resolve it.

Because router.refresh() didn't work for the dropdown, I resorted to using event listeners as a workaround. However, I'm curious to know why router.refresh() worked for other components but not for this dropdown. Is it because it's a client component? Does Next.js 13 offer a way to update client components specifically?

### Additional information

```js
**Code Snippets**

*AddFrequentIngreso.tsx*

// ... (rest of the code)
const handleAddFrequentIngreso = async () => {
  // ... (rest of the code)
  const event = new CustomEvent('frequentIngresoAdded');
  window.dispatchEvent(event);
  router.refresh();
};


*AddIngreso.tsx*
```typescript
// ... (rest of the code)
useEffect(() => {
  // ... (rest of the code)
  window.addEventListener('frequentIngresoAdded', fetchFrequentIngresos);
}, []);
```

**Dependencies**
- Next.js version: 13
- No error messages related to this issue
```

**Comment by devjiwonchoi** [comment, marked as answer]
Right, the `useRouter.refresh` in `next/navigation`, is to update Server Components without affecting the client side.

> Making a new request to the server, re-fetching data requests, and re-rendering Server Components. The client will merge the updated React Server Component payload without losing unaffected client-side React (e.g. useState) or browser state (e.g. scroll position).

Next.js currently does not directly support client-side updates and recommends using React Essentials.

### Reference
- [useRouter](https://nextjs.org/docs/app/api-reference/functions/use-router#userouter)

**Reply by vladimirzb** [reply]
Thanks for the reply!

**Comment by mo-cherif** [comment]
Got the same issue. as he said the router.refresh() is only available while using server components but you can try to use a react hook called useTransition then set the transition to router.refresh... works for me !

**Reply by shubhamkakkar** [reply]
How?

**Reply by IksGarG** [reply]
Also want to know this, how? 😂 @mo-cherif

**Reply by AnkitaMalik22** [reply]
Exactly! How??  We can use useTransition with router.refresh() if we're working in a server component right?
